packages/rum-core/src/performance-monitoring/metrics/inp/process.js

```diff
@@ -0,0 +1,256 @@
+   *
+   * One of the reasons why Google advocates for this is because it wouldn't be fair if your site would be penalized (SEO related),
+   * just because it received more interactions for an arbitrary reason.
+   */
+  const interactionIndex = Math.min(
+    inpState.longestInteractions.length - 1,
+    Math.floor(interactionCount() / 50)
+  )
+  const inp = inpState.longestInteractions[interactionIndex].duration
+
+  return inp
+}
+
+export function interactionCount() {
+  return performance.interactionCount || inpState.interactionCount
+}
+
+export function inpCandidates() {
+  return inpState.longestInteractions
+}
+
+export function setInteractionCount(count) {
+  inpState.interactionCount = count
+}
+
+export function restoreINPState() {
+  inpState.minInteractionId = Infinity
+  inpState.maxInteractionId = 0
+  inpState.interactionCount = 0
+  inpState.longestInteractions = []
+}
+
+// Stores user interaction
+function storeUserInteraction(entry) {
+  // Ignore interactions that are faster (or equal) than the least slow known interaction
+  const leastSlow =
+    inpState.longestInteractions[inpState.longestInteractions.length - 1]
+  if (
+    typeof leastSlow !== 'undefined' &&
+    entry.duration <= leastSlow.duration &&
+    entry.interactionId != leastSlow.id
+  ) {
+    return
+  }
+
+  /**
+   * different entries can share the same interactionId
+   * Please see the EventTiming spec for more details about
+   * the algorithm: https://www.w3.org/TR/event-timing/#sec-computing-interactionid
+   */
+  let foundIndex = -1
+  let i = 0
+  while (i < inpState.longestInteractions.length && foundIndex === -1) {
+    if (inpState.longestInteractions[i].id === entry.interactionId) {
+      foundIndex = i
+    }
+    i++
+  }
+
+  if (foundIndex >= 0) {
+    // Override existing interaction
+    inpState.longestInteractions[foundIndex].duration = Math.max(
+      inpState.longestInteractions[foundIndex].duration,
+      entry.duration
+    )
+  } else {
+    inpState.longestInteractions.push({
+      id: entry.interactionId,
+      duration: entry.duration
+    })
+  }
+
+  // Make sure we always keep the slowest interactions descendingly sorted
+  inpState.longestInteractions.sort((a, b) => b.duration - a.duration)
+  inpState.longestInteractions.splice(MAX_INTERACTIONS_TO_CONSIDER)
+}
+
+/**
+ * Updates the total count of different user interactions occurred in a page
+ * Once `performance.interactionCount` becomes globally available the "helper/polyfill logic" function
+ * will not be needed anymore.
+ *
+ * the reasons why interactionCount is important are:
+ * 1. To get the proper percentile 98 when calculating the INP value.
+ * 2. To report INP as 0 (rather than undefined) in those circumstances where the PerformanceObserver doesn't report entries.
+ *    That can happen because the min value for durationThreshold is 16 then interactions faster than that will not be reported.
```

Comment:
its a bit confusing, i think we can word it as `interactions faster than 16ms will not be taken in to account`

Reply:
I agree, besides there was a typo "it should slower than" :D